openvdb_points/tools/PointGroup.h

```diff
@@ -248,6 +259,59 @@ struct SetGroupFromIndexOp
 }; // struct SetGroupFromIndexOp
 
 
+template <typename PointDataTree, typename FilterT>
+struct SetGroupByFilterOp
+{
+    typedef typename tree::LeafManager<PointDataTree>   LeafManagerT;
+    typedef typename LeafManagerT::LeafRange            LeafRangeT;
+    typedef typename PointDataTree::LeafNodeType        LeafNodeT;
+    typedef AttributeSet::Descriptor::GroupIndex        GroupIndex;
+    typedef typename FilterT::Data                      FilterDataT;
+
+    SetGroupByFilterOp( const GroupIndex& index, const FilterDataT& filterData)
+        : mIndex(index)
+        , mFilterData(filterData) { }
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const
+    {
+        for (typename LeafManagerT::LeafRange::Iterator leaf=range.begin(); leaf; ++leaf) {
+
+            // obtain the group attribute array
+
+            GroupWriteHandle group(leaf->groupWriteHandle(mIndex));
+
+            // create the filter
+
+            FilterT filter(FilterT::create(*leaf, mFilterData));
+
+            // Use an IndexOnIter if a voxel Coord is required, otherwise an IndexIter will be faster
+
+            if (FilterTraits<FilterT>::RequiresCoord) {
+                typename LeafNodeT::IndexOnIter iter = leaf->beginIndexOn();
+                FilterIndexIter<typename LeafNodeT::IndexOnIter, FilterT> filterIndexIter(iter, filter);
+
+                for (; filterIndexIter; ++filterIndexIter) {
+                    group.set(*filterIndexIter, true);
+                }
+            }
+            else {
+                IndexIter iter = leaf->beginIndexAll();
+                FilterIndexIter<IndexIter, FilterT> filterIndexIter(iter, filter);
+
+                for (; filterIndexIter; ++filterIndexIter) {
+                    group.set(*filterIndexIter, true);
+                }
+            }
```

Comment:
Tried a different approach to address this, would be great to get your feedback.